File: src/main/java/de/grimsi/gameyfin/views/LoginView.java
package de.grimsi.gameyfin.views;

import com.vaadin.flow.component.UI;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.login.LoginForm;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.BeforeEnterEvent;
import com.vaadin.flow.router.BeforeEnterObserver;
import com.vaadin.flow.router.PageTitle;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.server.auth.AnonymousAllowed;
import de.grimsi.gameyfin.resources.PublicResources;
import de.grimsi.gameyfin.setup.SetupService;
import org.springframework.beans.factory.annotation.Autowired;


@Route("login")
@PageTitle("Login")
@AnonymousAllowed
public class LoginView extends VerticalLayout implements BeforeEnterObserver {

    private final LoginForm login = new LoginForm();

    public LoginView(@Autowired SetupService setupService) {
        if (!setupService.isSetupCompleted()) {
            UI.getCurrent().navigate(SetupView.class);
            UI.getCurrent().close();
        }
        Image logo = new Image(PublicResources.GAMEYFIN_LOGO.path, "Gameyfin");
        logo.setHeight("100px");

        login.setAction("login");

        add(logo);
        add(login);
    }

    @Override
    public void beforeEnter(BeforeEnterEvent beforeEnterEvent) {
        if (beforeEnterEvent.getLocation()
                .getQueryParameters()
                .getParameters()
                .containsKey("error")
        ) {
            login.setError(true);
        }
    }
}

File: src/main/kotlin/de/grimsi/gameyfin/ui/views/LoginView.kt
package de.grimsi.gameyfin.ui.views

import com.github.mvysny.karibudsl.v10.image
import com.github.mvysny.karibudsl.v10.loginForm
import com.vaadin.flow.component.login.LoginForm
import com.vaadin.flow.component.orderedlayout.FlexComponent
import com.vaadin.flow.component.orderedlayout.VerticalLayout
import com.vaadin.flow.router.BeforeEnterEvent
import com.vaadin.flow.router.BeforeEnterObserver
import com.vaadin.flow.router.PageTitle
import com.vaadin.flow.router.Route
import com.vaadin.flow.server.auth.AnonymousAllowed
import de.grimsi.gameyfin.ui.resources.PublicResources

@Route("login")
@PageTitle("Login")
@AnonymousAllowed
class LoginView : VerticalLayout(), BeforeEnterObserver {

    private var login: LoginForm

    init {
        setSizeFull()
        justifyContentMode = FlexComponent.JustifyContentMode.CENTER
        alignItems = FlexComponent.Alignment.CENTER

        image {
            height = "100px"
            src = PublicResources.GAMEYFIN_LOGO.path
            setAlt("Gameyfin")
        }

        login = loginForm {
            addClassName("login-view")
            action = "login"
        }
    }

    override fun beforeEnter(event: BeforeEnterEvent?) {
        if (event != null) {
            if (event.location
                    .queryParameters
                    .parameters
                    .containsKey("error")
            ) {
                login.isError = true
            }
        }
    }

}